platform/kube/controller.go

```diff
@@ -251,6 +252,21 @@ func (c *Controller) appendIngressConfigHandler(k string, f func(model.Key, prot
 			return nil
 		}
 
+		// TODO: priority?
+		for _, tls := range ingress.Spec.TLS {
+			if tls.SecretName == "" {
+				continue
+			}
+
+			if len(tls.Hosts) == 0 {
+				c.config.Secrets.put("*", tls.SecretName)
+			} else {
+				for _, host := range tls.Hosts {
+					c.config.Secrets.put(host, tls.SecretName)
+				}
+			}
+		}
+
```

Comment:
Yes, there are two event notifications that need to be handled: 1) updating the secret storage map (user changing the ingress resource may change the host secret 2) restarting the proxy if the content of the secret changes